skbio/tree/_tree.py

```diff
@@ -617,6 +617,72 @@ def lowest_common_ancestor(self, tipnames):
 
     lca = lowest_common_ancestor  # for convenience
 
+    def path(self, other, include_ends=False):
+        r"""Return the list of node names in the path from one node to another.
+
+        Parameters
+        ----------
+        other : TreeNode
+            Final node of path.
+        include_ends: bool, optional
+            Whether to include the initial and final nodes in the list.
+
+        Returns
+        -------
+        list
+            List of names of TreeNode objects.
+
+        Examples
+        --------
+        >>> from skbio import TreeNode
+        >>> tree = TreeNode.read(["((a,b)c,(d,e)f)root;"])
+        >>> node_1, node_2 = tree.find('a'), tree.find('d')
+        >>> path = node_1.path(node_2)
+        >>> print(path)
+        ['c', 'root', 'f']
+        >>> path_2 = node_1.path(node_2, include_ends=True)
+        >>> print(path_2)
+        ['a', 'c', 'root', 'f', 'd']
+
+        """
+        # find root
+        root = self.root()
+
+        # find the lowest common ancestor if not ancestors of each other
+        lca = (
+            self
+            if self in other.ancestors()
+            else other
+            if other in self.ancestors()
+            else root.lowest_common_ancestor([self, other])
+        )
+
+        # create tree rooted at lca
+        lca_tree = lca.copy()
```

Comment:
Copying a subtree is an expensive operation. We should avoid it wherever possible. Additionally, once you copy a tree, you lose reference to original nodes.